src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -0,0 +1,348 @@
+import React, { useEffect, useRef, useState } from "react";
+import SummaryCard from "../../ui-components/cards/summary-card.component";
+import styles from "./immunizations-form.css";
+import { DataCaptureComponentProps } from "../shared-utils";
+import { useTranslation } from "react-i18next";
+import { savePatientImmunization } from "./immunizations.resource";
+import { mapToFhirImmunizationResource } from "./immunization-mapper";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import { useHistory } from "react-router-dom";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { getStartedVisit } from "../visit/visit-utils";
+import useSessionUser from "../../utils/use-session-user";
+
+export function ImmunizationsForm(props: ImmunizationsFormProps) {
+  const [vaccineName, setVaccineName] = useState("");
+  const [vaccineUuid, setVaccineUuid] = useState("");
+  const [immunizationObsUuid, setImmunizationObsUuid] = useState("");
+  const [vaccinationDate, setVaccinationDate] = useState(null);
+  const [immunizationSequences, setImmunizationSequences] = useState([]);
+  const [currentDose, setCurrentDose] = useState<ImmunizationSequence>(
+    {} as ImmunizationSequence
+  );
+  const [vaccinationExpiration, setVaccinationExpiration] = useState(null);
+  const [lotNumber, setLotNumber] = useState("");
+  const [manufacturer, setManufacturer] = useState("");
+  const [enableCreateButtons, setEnableCreateButtons] = useState(false);
+  const [enableEditButtons, setEnableEditButtons] = useState(true);
+  const [viewEditForm, setViewEditForm] = useState(true);
+  const [formChanged, setFormChanged] = useState<Boolean>(false);
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+  const formRef = useRef<HTMLFormElement>(null);
+  const { t } = useTranslation();
+  const history = useHistory();
+  const today = new Date().toISOString().split("T")[0];
+  const currentUser = useSessionUser();
+
+  useEffect(() => {
+    if (vaccinationDate) {
+      setEnableCreateButtons(true);
+    } else {
+      setEnableCreateButtons(false);
+    }
+  }, [vaccinationDate]);
+
+  useEffect(() => {
+    if (viewEditForm && formChanged) {
+      setEnableEditButtons(true);
+    } else {
+      setEnableEditButtons(false);
+    }
+  }, [viewEditForm, formChanged]);
+
+  useEffect(() => {
+    if (props.match.params) {
+      const {
+        immunizationObsUuid,
+        vaccineName,
+        vaccineUuid,
+        manufacturer,
+        expirationDate,
+        vaccinationDate,
+        lotNumber,
+        sequences,
+        currentDose
+      }: Immunization = props.match.params[0];
```

Comment:
Does this work? Usually the `match` params are always strings. Also I don't know if you can access them by number - usually they use their id, e.g., in a route definition `/foo/:name` the params of `/foo/flo` would be `{ name: 'flo' }`.

Maybe this is something different and I don't understand what `match` should be in this case (or `ImmunizationsFormProps` in general). Can we type this better?

Reply:
Let me give it a try to better type it. However the code we have referred `ConditionForm` for a lot of initial things. We will see if it could be better represented.

Reply:
> Does this work? Usually the match params are always strings.
If I understand it clearly, you are referring to the part `props.match.params[0];`. This was a miss on our side. We were sending params as an array with one item. We have fixed it.